### ISSUE TITLE
Linear Regression

### ISSUE THREAD

**Issue by bvpav** [CodenameSource/libuaic#6]
Имплементиране на множествена линейна регресия

# Структура на линейната регресия
 - ![image](https://github.com/CodenameSource/libuaic/assets/65411104/814b72ca-0f81-400e-93ea-59ebc2bca40c)
  x - Матрица с `feature` променливите, w - Матрица с коефициентите, b - отклонение

- [x] Структура с коефициенти
- [x] Loss function, фунцкия калкулираща грешката на предсказания, от алгоритъма, резултат от реалния (За едно поле от тренировъчните данни)
![image](https://github.com/CodenameSource/libuaic/assets/65411104/cd6e987d-d51e-4e6e-895f-111c64272d11)
Minimum squared error
- [x] Optimizer function - Gradient descent, оптимизираща функция
- [x] Fit function, функция която тренира алгоритъма върху въведените данни
- [x] Predict function, функция която взима коефициентите и връща предсказания резултат спрямо въведените данни на базата на тренираните коефициенти

# Ресурси

- [Linear regression from scratch](https://developer.ibm.com/articles/linear-regression-from-scratch/)
- [Linear Regression with and without numpy](https://commandlinefanatic.com/cgi-bin/showarticle.cgi?article=art084)
- [Multiple linear regression](https://betterdatascience.com/mml-multiple-linear-regression/)
- [Gradient descent](https://philippmuens.com/gradient-descent-from-scratch)
